### DISCUSSION TITLE
Utilize server side framework Vert.x to provide FXGL game servers?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1051] [Ideas]
Hi:

[Vert.x is a Eclipse foundation project](https://vertx.io/) and it provides Actor like frameworks on the server side
We used to use Vert.x and JavaFX to create online games
Just a thought, is it possible to provide FXGL server module for Vert.x ?
Since a lot of concepts are similar e.g. Eventbus, EventLoop etc.
and the nice part of Vert.x is it provide a elegant way of avoiding multiple thread programming
just use Verticle which is a component bind to a loop thread(eventloop) and use eventbus to joint them
so is it possible to create a project to facilitate the development especially the network communication between client created by FXGL and the server created by Vert.x?

Thanks

**Comment by AlmasB** [comment]
Hi, I'm not familiar with the concepts in Vert.x, so I'm not able to answer this question. Someone with knowledge of both vert.x and fxgl (or javafx) would be more suited to answer this.
